cmd/otel-agent/config/agent_config.go

```diff
@@ -0,0 +1,122 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package config provides a way to convert the OpenTelemetry Collector configuration to the Datadog Agent configuration.
+package config
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/datadogexporter"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/converter/expandconverter"
+	"go.opentelemetry.io/collector/confmap/provider/envprovider"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
+	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
+	"go.opentelemetry.io/collector/service"
+)
+
+// NewConfigComponent creates a new config component from the given URIs
+func NewConfigComponent(ctx context.Context, uris []string) (config.Component, error) {
+	// Load the configuration from the fileName
+	rs := confmap.ResolverSettings{
+		URIs: uris,
+		ProviderFactories: []confmap.ProviderFactory{
+			fileprovider.NewFactory(),
+			envprovider.NewFactory(),
+			yamlprovider.NewFactory(),
+			httpprovider.NewFactory(),
+			httpsprovider.NewFactory(),
+		},
+		ConverterFactories: []confmap.ConverterFactory{expandconverter.NewFactory()},
+	}
+
+	resolver, err := confmap.NewResolver(rs)
+	if err != nil {
+		return nil, err
+	}
+	cfg, err := resolver.Resolve(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ddc, err := getDDExporterConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	sc, err := getServiceConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	site := ddc.API.Site
+	apiKey := string(ddc.API.Key)
```

Comment:
There is work being done in the core-agent to allow refreshing the API Key without restarting. We haven't discussed how the Otel-Agent fits into that, but if it wants the same functionality, we'll eventually need to add something here to get change notifications from the config and update this value that has been copied out.

Reply:
Good call out. In future we should fetch API key/ config component from core Agent.